theia-tree-editor/src/browser/detail-form-widget.tsx

```diff
@@ -36,111 +33,100 @@ const JSON_FORMS_CONTAINER_CSS_CLASS = 'jsonforms-container';
-            initState
-        );
-    }
-
-    /** Augments the default vanilla styles with theia-specific styling and returns the result. */
-    private createStyles(): StyleDef[] {
-        const registerStylesAction = registerStyles([
-            {
-                name: 'array.button',
-                classNames: ['theia-button']
-            },
-            {
-                name: 'array.table.button',
-                classNames: ['theia-button']
-            },
-            {
-                name: 'control.input',
-                classNames: ['theia-input']
-            },
-            {
-                name: 'control.select',
-                classNames: ['theia-select']
-            }
-        ]);
-        return stylingReducer(vanillaStyles, registerStylesAction);
-    }
-
     setSelection(selectedNode: TreeEditor.Node): void {
         this.selectedNode = selectedNode;
-
-        this.store.dispatch(
-            Actions.init(
-                this.modelService.getDataForNode(this.selectedNode),
-                this.modelService.getSchemaForNode(this.selectedNode),
-                this.modelService.getUiSchemaForNode(this.selectedNode),
-                {
-                    refParserOptions: {
-                        dereference: { circular: 'ignore' }
-                    },
-                    useDefaults: true
-                }
-            )
-        );
         this.renderForms();
     }
 
     protected renderForms(): void {
         if (this.selectedNode) {
+            const data = this.modelService.getDataForNode(this.selectedNode);
+            const schema = this.modelService.getSchemaForNode(this.selectedNode);
+            const uiSchema = this.modelService.getUiSchemaForNode(this.selectedNode);
+
             ReactDOM.render(
                 <div className={JSON_FORMS_CONTAINER_CSS_CLASS}>
-                    <Provider store={this.store}>
-                        <JsonFormsReduxContext>
-                            <JsonFormsDispatch />
-                        </JsonFormsReduxContext>
-                    </Provider>
+                    <JsonFormsStyleContext.Provider value={styleContextValue}>
+                        <JsonForms
+                            data={data}
+                            schema={schema}
+                            uischema={uiSchema}
+                            cells={vanillaCells}
+                            renderers={vanillaRenderers}
+                            onChange={this.jsonformsOnChange}
+                            config={{
+                                restrict: false,
+                                trim: false,
+                                showUnfocusedDescription: true,
+                                hideRequiredAsterisk: false
+                            }}
+                            refParserOptions={{
+                                dereference: { circular: 'ignore' }
+                            }}
```

Comment:
`cells`, `renderers`, `config` and `refParserOptions` could be retrieved by a protected method, e.g. `getJsonFormsConfig` which is then added here via `{...getJsonFormsConfig()}`. Then we would be fully flexible. Of course the method should always return the same elements.